formats/sarifutils/sarifutils.go

```diff
@@ -19,12 +17,30 @@ func NewReport() (*sarif.Report, error) {
 	return report, nil
 }
 
-func ConvertSarifReportToString(report *sarif.Report) (sarifStr string, err error) {
-	out, err := json.Marshal(report)
-	if err != nil {
-		return "", errorutils.CheckError(err)
+func NewPhysicalLocation(physicalPath string) *sarif.PhysicalLocation {
+	return &sarif.PhysicalLocation{
+		ArtifactLocation: &sarif.ArtifactLocation{
+			URI: &physicalPath,
+		},
+	}
+}
+
+func NewPhysicalLocationWithRegion(physicalPath string, startRow, endRow, startCol, endCol int) *sarif.PhysicalLocation {
+	location := NewPhysicalLocation(physicalPath)
+	location.Region = &sarif.Region{
+		StartLine:   &startRow,
+		EndLine:     &endRow,
+		StartColumn: &startCol,
+		EndColumn:   &endCol,
+	}
+	return location
+}
+
+func NewLogicalLocation(name, kind string) *sarif.LogicalLocation {
```

Comment:
same as `NewPhysicalLocationWithRegion` - need to pass by ref?

Reply:
yes, this is required by the library we use, it holds pointers